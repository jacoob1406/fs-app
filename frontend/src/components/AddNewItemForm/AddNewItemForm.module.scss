.addItemContainer {
  display: flex;
  justify-content: center;
  align-items: center;

  &__heading {
    margin: 0 0.5rem 0.5rem;
  }

  &__form {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    &--item {
      display: grid;
      grid-template-columns: 100px 150px;
      grid-gap: 5px;

      label,
      select,
      input {
        margin: 0 0 0.3rem;
        padding: 10px;
        font-size: 1rem;
      }
    }
  }

  &__button {
    border: none;
    font-size: 1.2rem;
    background: #ffc703;
    cursor: pointer;
    border-radius: 3px;
    padding: 2px 15px;
    margin: 10px;

    &:hover {
      color: #ffffff;
    }
    &:focus {
      outline: none;
    }
    &:disabled {
      background: #585858;
      color: #ffffff;
      cursor: not-allowed;
    }
  }
}

@media (max-width: 560px) {
  .addItemContainer {
    flex-direction: column;
  }
}
