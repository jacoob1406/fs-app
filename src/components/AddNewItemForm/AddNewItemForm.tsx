import React, { useState, FC } from 'react';
import styles from './AddNewItemForm.module.scss';
import { AddNewItemBody } from '../../containers/AddNewItem/model/AddNewItemAction';

interface Props {
  handleSubmit(values: AddNewItemBody): void;
}

const AddNewItemForm: FC<Props> = ({ handleSubmit }: Props) => {
  const [category, setCategory] = useState<string>('Career');
  const [description, setDescription] = useState<string>('');

  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) =>
    setCategory(event.currentTarget.value);

  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    setDescription(event.currentTarget.value);

  const isDescriptionEmpty = (): boolean => description === '';

  const submitAddItem = (event: React.MouseEvent | React.FormEvent) => {
    if (isDescriptionEmpty()) {
      event.preventDefault();
      return;
    }
    handleSubmit({ category: category, text: description });
    setDescription('');
  };

  return (
    <section className={styles.addItemContainer}>
      <h2 className={styles.addItemContainer__heading}>Add new item:</h2>
      <form onSubmit={submitAddItem} className={styles.addItemContainer__form}>
        <div className={styles['addItemContainer__form--item']}>
          <label>Category:</label>
          <select onChange={handleSelect} value={category}>
            <option value='Career'>Career</option>
            <option value='Family'>Family</option>
            <option value='Travelling'>Travelling</option>
          </select>
        </div>
        <div className={styles['addItemContainer__form--item']}>
          <label>Text:</label>
          <input type='text' value={description} onChange={handleInputChange} />
        </div>
      </form>
      <button
        className={styles.addItemContainer__button}
        onClick={submitAddItem}
        disabled={isDescriptionEmpty()}
      >
        Add
      </button>
    </section>
  );
};

export default AddNewItemForm;
