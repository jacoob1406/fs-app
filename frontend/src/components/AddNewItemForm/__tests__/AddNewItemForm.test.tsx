import React from 'react';
import { render } from '@testing-library/react';
import AddNewItemForm from '../AddNewItemForm';

describe('AddNewItemForm component', () => {
  it('should match snapshot', () => {
    const component = render(<AddNewItemForm handleSubmit={() => undefined} />);
    expect(component.container).toMatchSnapshot();
  });
});
