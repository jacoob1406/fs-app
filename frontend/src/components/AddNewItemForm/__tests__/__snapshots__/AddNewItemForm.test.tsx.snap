// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewItemForm component should match snapshot 1`] = `
<div>
  <section
    class="addItemContainer"
  >
    <h2
      class="addItemContainer__heading"
    >
      Add new item:
    </h2>
    <form
      class="addItemContainer__form"
    >
      <div
        class="addItemContainer__form--item"
      >
        <label>
          Category:
        </label>
        <select>
          <option
            value="Career"
          >
            Career
          </option>
          <option
            value="Family"
          >
            Family
          </option>
          <option
            value="Travelling"
          >
            Travelling
          </option>
        </select>
      </div>
      <div
        class="addItemContainer__form--item"
      >
        <label>
          Text:
        </label>
        <input
          type="text"
          value=""
        />
      </div>
    </form>
    <button
      class="addItemContainer__button"
      disabled=""
    >
      Add
    </button>
  </section>
</div>
`;
